src/operators/ExtrudeStacked/QvisExtrudeStackedWindow.C

```diff
@@ -0,0 +1,857 @@
+        if(item->text(0) == oldVariable)
+        {
+            variableTree->setCurrentItem(item);
+            variableSelected(item);
+            found = true;
+        }
+    }
+    if (!found && nitems > 0)
+    {
+        QTreeWidgetItem *item = variableTree->topLevelItem(0);
+        variableTree->setCurrentItem(item);
+        variableSelected(item);
+    }
+}
+
+
+// ****************************************************************************
+// Method: QvisExtrudeStackedWindow::GetCurrentValues
+//
+// Purpose:
+//   Gets values from certain widgets and stores them in the subject.
+//
+// Note:       Autogenerated by xml2window.
+//
+// Programmer: xml2window
+// Creation:   omitted
+//
+// Modifications:
+//
+// ****************************************************************************
+
+void
+QvisExtrudeStackedWindow::GetCurrentValues(int which_widget)
+{
+    bool doAll = (which_widget == -1);
+
+    // Do axis
+    if(which_widget == ExtrudeStackedAttributes::ID_axis || doAll)
+    {
+        double val[3];
+        if(LineEditGetDoubles(axis, val, 3))
+            atts->SetAxis(val);
+        else
+        {
+            ResettingError(tr("Extrusion axis"),
+                DoublesToQString(atts->GetAxis(),3));
+            atts->SetAxis(atts->GetAxis());
+        }
+    }
+
+    // Do length
+    if(which_widget == ExtrudeStackedAttributes::ID_length || doAll)
+    {
+        double val;
+        if(LineEditGetDouble(length, val))
+            atts->SetLength(val);
+        else
+        {
+            ResettingError(tr("Length"),
+                DoubleToQString(atts->GetLength()));
+            atts->SetLength(atts->GetLength());
+        }
+    }
+
+    // Do steps
+    if(which_widget == ExtrudeStackedAttributes::ID_steps || doAll)
+    {
+        int val;
+        if(LineEditGetInt(steps, val))
+            atts->SetSteps(val);
+        else
+        {
+            ResettingError(tr("Number of steps"),
+                IntToQString(atts->GetSteps()));
+            atts->SetSteps(atts->GetSteps());
+        }
+    }
+}
+
+
+//
+// Qt Slot functions
+//
+
+
+void
+QvisExtrudeStackedWindow::axisProcessText()
+{
+    GetCurrentValues(ExtrudeStackedAttributes::ID_axis);
+    Apply();
+}
+
+void
+QvisExtrudeStackedWindow::lengthProcessText()
+{
+    GetCurrentValues(ExtrudeStackedAttributes::ID_length);
+    Apply();
+}
+
+
+void
+QvisExtrudeStackedWindow::stepsProcessText()
+{
+    GetCurrentValues(ExtrudeStackedAttributes::ID_steps);
+    Apply();
+}
+
+
+void
+QvisExtrudeStackedWindow::preserveOriginalCellNumbersChanged(bool val)
+{
+    atts->SetPreserveOriginalCellNumbers(val);
+    SetUpdate(false);
+    Apply();
+}
+
+// ****************************************************************************
+//  Method:  QvisExtrudeStackedWindow::variableGroupChanged
+//
+//  Purpose:
+//    extrude by variable(s)
+//
+//  Arguments:
+//    val   boolean true - extrude by variable
+//
+//  Programmer:  Jeremy Meredith
```

Comment:
Make yourself the programmer, use current Creation date. If copied or pulled from another place, make a Note stating so.
Here and elsewhere in the file.

Reply:
Done